tutorials/RandomNumberGeneration/Tests.qs

```diff
@@ -17,65 +17,108 @@ namespace Quantum.Kata.RandomNumberGeneration {
         return RandomBit();
     }
 
 
     // Exercise 2.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T2_RandomTwoBits () : Unit {
-        Message("Testing...");
-        CheckFlatDistribution(RandomTwoBits_Wrapper, 2, 1.4, 1.6, 1000, 200);
+        Message("Testing Two RandomBitGeneration...");
+        RetryCheckUniformDistribution(RandomTwoBits_Wrapper, 0, 3, 1000, 3);
+        Message("Two random bits successfully generated...");
     }
 
-    operation RandomTwoBits_Wrapper (throwaway: Int) : Int {
+    operation RandomTwoBits_Wrapper (throwawayMin: Int, throwawayMax: Int) : Int {
         return RandomTwoBits();
     }
     
 
     // Exercise 3.
-    @Test("QuantumSimulator")
+    @Test("Microsoft.Quantum.Katas.CounterSimulator")
     operation T3_RandomNBits () : Unit {
         Message("Testing N = 1...");
-        CheckFlatDistribution(RandomNBits, 1, 0.4, 0.6, 1000, 450);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1, 1000, 3);
+        Message("Test Passed Succesfully for N=1 \n");
+
         Message("Testing N = 2...");
-        CheckFlatDistribution(RandomNBits, 2, 1.4, 1.6, 1000, 200);
-        Message("Testing N = 3...");
-        CheckFlatDistribution(RandomNBits, 3, 3.3, 3.7, 1000, 90);
-        Message("Testing N = 10...");
-        CheckFlatDistribution(RandomNBits, 10, 461.0, 563.0, 1000, 0);
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 3, 1000, 3);
+        Message("Test Passed Succesfully for N=2 \n");
+
+        Message("\n Testing N = 3...");
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 7, 1000, 3);
+        Message("Test Passed Succesfully for N=3 \n");
+
+        Message("\n Testing N = 10...");
+        RetryCheckUniformDistribution(RandomNBits_Wrapper, 0, 1023, 1000, 3);
+        Message("Test Passed Succesfully for N=10 \n");
+    }
+
+    operation RandomNBits_Wrapper (min: Int, max: Int) : Int {
+        // For N bit random number : min = 0, max = 2^N - 1
+        let N = Ceiling( Lg(IntAsDouble(max+1)) );
+        return RandomNBits(N);
+    }
+
+    operation RetryCheckUniformDistribution( fTest: ((Int,Int)=>Int), min: Int, max: Int, nRuns: Int, numRetries: Int) : Unit {
+        mutable sufficientlyRandom = false;
+        mutable attemptNum = 1;
+        repeat
+        {
+            Message($"attemptNum = {attemptNum}");
+            set sufficientlyRandom = CheckUniformDistribution(fTest, min, max, nRuns);
+            set attemptNum += 1;
+        } until( sufficientlyRandom == true or attemptNum >= numRetries );
+
+        if(sufficientlyRandom == false)
+        {
+            fail $"Failed to generate a random integer as required";
+        }
     }
 
     /// # Input
     /// ## f
     /// Random number generation operation to be tested
     /// ## numBits
     /// Number of bits in the generated result
-    /// ## lowRange
```

Comment:
Please update the documentation to match the inputs; `numBits` is no longer a parameter but `min` and `max` are.

Reply:
Done